### ISSUE TITLE
Accessibility issue: duplicate links on example for summary cards

### ISSUE THREAD

**Issue by frankieroberto** [alphagov/govuk-design-system#2680]
## Description of the issue

The [example for the summary cards feature](https://design-system.service.gov.uk/components/summary-list/card-with-title/index.html) has "Change" links with visually hidden text to make them unique within each section, for example `Change [age]` but these links are not unique across the page as a whole.

Should they be amended to say `Change [thing] for [person name]`?

Alternatively, is the heading and the rest of the summary card sufficient to give enough context for the link?

Related WCAG criteria: [Success Criterion 2.4.9 Link Purpose (Link Only)](https://www.w3.org/TR/WCAG21/#link-purpose-link-only)

**Comment by davidc-gds** [comment]
Hi @frankieroberto, thanks for submitting this! We've had a bit of a chat within the team, and think you may have caught something — it seems fishy. 🐟 

Here's my initial assessment of the WCAG-side of things. This assessment would benefit greatly from second opinions, if we can find them.

## Initial assessment
This likely fails [2.4.4. Link Purpose (In Context) (A)](https://www.w3.org/TR/WCAG22/#link-purpose-in-context), though only just barely, and due to a technicality. 😕 

Since that's a Level A criterion, we can focus on that one for now, instead of [2.4.9 Link Purpose (Link Only) (AAA)](https://www.w3.org/TR/WCAG22/#link-purpose-link-only)

**Note:** I would want to confirm the following analysis with other groups, like the accessibility monitoring team, if we can.

The key thing to consider is that ‘programmatically determined link context’ is a specifically defined term in WCAG. Here’s the [definition for ‘programmatically determined link context’](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html#dfn-programmatically-determined-link-context), from the Link Purpose 2.4.4 ‘Understanding’ page.

> ### programmatically determined link context
> additional information that can be [programmatically determined](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html#dfn-programmatically-determined) from [relationships](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html#dfn-relationships) with a link, combined with the link text, and presented to users in different modalities
> In HTML, information that is programmatically determinable from a link in English includes text that is in the same paragraph, list, or table cell as the link or in a table header cell that is associated with the table cell that contains the link.

So since the ‘link context’ in this case comes from the summary card heading, which isn’t programmatically determinable via any direct relationship with the link, it fails. Even if HTML elements are located directly beside each other (like a `<div>` beside an `<a>`), they aren’t considered programmatically determined link text.

## Potential fixes
There’s 2 ways to fix this (I think?):

- Add `aria-describedby` to connect each summary card’s heading to its ‘Change’ links.
    - I may be wrong, but I don’t think we can use `aria-labelledby` because that’ll override the inherent label provided by the text in the link (which is `Change<span class="govuk-visually-hidden"> age</span>`).
- Add the text from the summary card’s heading to each of the hidden span text pieces. That takes a bit more effort to add each one, but then at least we’re adding all the link context in one place.

**Comment by frankieroberto** [comment]
@davidc-gds someone asked me about whether we’d assessed the accessibility of the summary cards (but not specifically the change links) via the x-gov Slack, and I spotted it then.

In the [Apply for teacher training live service](http://www.apply-for-teacher-training.service.gov.uk/candidate/account) we use the summary card (we’ve not yet migrated to the version in govuk-frontend, but it’s very similar).

Where we have change links with summary cards, we’ve added extra visually-hidden text to the change links to make sure they're unique, for example `Change [year for A level, Biology, 2016]` or `Change [email for Joe Bloggs]` or `Change [grade for Bachelor of Applied Science, Environmental history, University of Dundee, 2021]`. Brackets denote visually hidden text.

I don’t think this was done in response to any direct testing with users of assistive software, but was done in anticipating that unique links might be required. The service was audited by the Digital Accessibility Centre, who didn’t raise any issues about this directly (but did report a separate issue where there were non-unique change links).

**Comment by davidc-gds** [comment]
Thanks for the extra context, @frankieroberto! 

Based on that info, I’m going to suggest this is a medium-severity accessibility concern, despite being a potential WCAG failure.

There’s little evidence it’s affecting service teams or people using the services.

Here's the [factors we use to determine the severity of an accessibility concern](https://design-system.service.gov.uk/community/accessibility-strategy/#determining-severity-of-an-accessibility-concern):

- there is reasonable evidence that it makes it difficult or impossible for some people to complete certain tasks:
    - Not that we know of at this point :x:
- it affects the accessibility of essential services or critical infrastructure:
    - It's possible, but there's no evidence of it at this time :x:
- individual service teams cannot efficiently resolve it on their own:
    - This is fairly easy for individual teams to fix, if they catch it :x:

If it wasn't failing a WCAG level A criterion I would consider marking this as a low-severity issue. But since there's WCAG to consider, that bumps it up to medium in my mind.

**Comment by davidc-gds** [comment]
Hi all, I've submitted a proposal to add this as a listed piece of non-accessible content in our [accessibility statement](https://design-system.service.gov.uk/accessibility/). It would be an issue with the Design System website and not with GOV.UK Frontend, because it's specifically an issue of how we implement our examples on the [Summary list guidance page](https://design-system.service.gov.uk/components/summary-list/).

**Comment by davidc-gds** [comment]
Hey @frankieroberto we think this might be resolved by an upcoming V5 change:
- https://github.com/alphagov/govuk-frontend/issues/3674

And similarly, the issue will be fixed for summary card actions as well:
- https://github.com/alphagov/govuk-frontend/issues/3674

We'll leave this listed in the [accessibility statement non-compliance section](https://design-system.service.gov.uk/accessibility/#non-compliance-with-the-accessibility-regulations) until V5 is released though, just so it's tracked.

**Comment by michael-johnson-opencast** [comment]
Hey everyone, this is a really interesting thread and I'd like to add something to the conversation that you may not have considered. When a user chooses to use a screen reader, they can set their device to "skim read" a page by the type of thing that they're looking for, for example, headings, links, forms, buttons, etc. If links don't contain information as to what the outcome will be of clicking on them then "skim reading" won't make any sense. For example, if you choose to just write "Change" as a link, it doesn't mean anything. The best way to write a link is to give context, so as you've already suggested, "Change [Email for Joe Bloggs]". I would advise [as an accessibility specialist] to avoid using Aria to supersede HTML wherever possible. The reason is that screen reader users get used to how things are announced by the operating system that their device uses and if you supersede that with something that's unconventional it may create confusion. I hope this helps.
